src/Orchard.Web/Modules/Orchard.Setup/Services/WebconfigOverrides.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Configuration;
+
+namespace Orchard.Setup.Services
+{
+
+    public class WebconfigOverrides
+    {
+        private const string AppSettingDefaultDataProvider = "defaultProvider";
+        private const string AppSettingDefaultRecipe = "defaultRecipe";
+        private const string ConnectionStringDefaultConnectionString = "defaultConnection";
+
+        public string DataProvider { get; set; }
+        public string DataConnectionString { get; set; }
+        public string DefaultRecipe { get; set; }
+
+        public static WebconfigOverrides Load()
+        {
+            var result = new WebconfigOverrides();
+
+            if (ConfigurationManager.ConnectionStrings[ConnectionStringDefaultConnectionString] != null)
+            {
+                result.DataConnectionString = ConfigurationManager.ConnectionStrings[ConnectionStringDefaultConnectionString].ConnectionString;
```

Comment:
You need to prefix with the Tenant's name like `tenantName + ":" + ConnectionStringDefaultConnectionString` and fallback to one without the tenant name. Because otherwise this would override the value for subsequent tenants.